### PR TITLE
Improve `malloc_object_size` by calculating offset from start for small allocations

### PR DESCRIPTION
**Pull request by rwarr627**
Addresses issue #23

### CODE REVIEW

**Comment by thestinger** [comment]
Can you rebase this so there isn't a merge?

**Comment by thestinger** [comment]
@rwarr627 Just need you to rebase this to deal with the conflict(s) and clean up the history.

File: README.md [hunk 1]
```diff
@@ -796,14 +796,7 @@ The `size_t malloc_object_size(void *ptr)` function returns an *upper bound* on
 the accessible size of the relevant object (if any) by querying the malloc
 implementation. It's similar to the `__builtin_object_size` intrinsic used by
 `_FORTIFY_SOURCE` but via dynamically querying the malloc implementation rather
-than determining constant sizes at compile-time. The current implementation is
-just a naive placeholder returning much looser upper bounds than the intended
-implementation. It's a valid implementation of the API already, but it will
-become fully accurate once it's finished. This function is **not** currently
```
Comment (thestinger) [inline README.md:802]: The documentation on signal safety should be left here since this hasn't been addressed like before yet.

File: README.md [hunk 2]
```diff
@@ -796,14 +796,7 @@ The `size_t malloc_object_size(void *ptr)` function returns an *upper bound* on
 the accessible size of the relevant object (if any) by querying the malloc
 implementation. It's similar to the `__builtin_object_size` intrinsic used by
 `_FORTIFY_SOURCE` but via dynamically querying the malloc implementation rather
-than determining constant sizes at compile-time. The current implementation is
```
Comment (thestinger) [inline README.md:799]: It's still not fully accurate for large allocations. I'd just leave this documentation alone and handle this in a separate PR.

**Comment by thestinger** [comment]
Start with `git rebase upstream/master` to rebase onto the current master (assuming you have this repository as a remote named `upstream`). Then, do an interative rebase (`git rebase -i PARENT_BEFORE_YOUR_CHANGES`) to remove the documentation change for the time being (I think it should just be done separately).

You might as well also squash the `LDFLAGS` commit into the one adding the tests (change `pick` to `squash`).

**Comment by rwarr627** [comment]
Okay I think I've made the necessary adjustments.

**Pull request merged**